src/test/java/com/openlattice/rehearsal/search/SearchControllerTest.kt

```diff
@@ -373,12 +411,104 @@ class SearchControllerTest : MultipleAuthenticatedUsersBase() {
         /* Soft delete */
         dataApi.deleteAllEntitiesFromEntitySet(es2.id, DeleteType.Soft)
 
-        // should be un-indexed in background job
-        Thread.sleep(30_000L)
+        // should be deleted automatically
         searchedEntities1 = searchApi.executeEntitySetDataQuery(es1.id, searchAll)
         searchedEntities2 = searchApi.executeEntitySetDataQuery(es2.id, searchAll)
         Assert.assertEquals(0, searchedEntities1.numHits)
         Assert.assertEquals(0, searchedEntities2.numHits)
     }
 
+    @Test
+    fun testMaxHitsSearch() {
+        // create 10 entities
+        val et = createEntityType()
+        val es = createEntitySet(et)
+
+        val numberOfEntities = 20
+        val testData = TestDataFactory.randomStringEntityData(numberOfEntities, et.properties).values.toList()
+        dataApi.createEntities(es.id, testData).toSet().zip(testData).toMap()
+        Thread.sleep(1000)
+
+        val searchMax0 = SearchTerm("*", 0, 0)
+        var searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax0)
+
+        // should yield 0 hits and 0 totalHits
+        Assert.assertEquals(0, searchedEntities.hits.size)
+        Assert.assertEquals(numberOfEntities, searchedEntities.numHits.toInt())
+
+        val searchMax1 = SearchTerm("*", 0, 1)
+        searchedEntities = searchApi.executeEntitySetDataQuery(es.id, searchMax1)
+
+        // should yield 1 hit, but 10 totalHits
```

Comment:
tiny issue but the 10 in this comment is incorrect. :)

Reply:
Right.. thanks! I updated `numberOfEntities` to 20, but then I made a different test for deleting + maxHit limited search. Will change it back to 10